### PR TITLE
Fix "first Friday of next month" returning Friday of next week

### PR DESCRIPTION
**Pull request by AniChikage**

### CODE REVIEW

File: .NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDateParser.cs [hunk 1]
```diff
@@ -627,7 +627,7 @@ private static DateObject ComputeDate(int cardinal, int weekday, int month, int
                 weekday = 7;
             }
 
-            if (weekday < (int)firstDay.DayOfWeek)
+            if (weekday < ((int)firstDay.DayOfWeek == 0 ? 7 : (int)firstDay.DayOfWeek))
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDateParser.cs]: Please assign the value to a variable and use it in the if condition.
Also, please add test cases, to show that the bug was really fixed and that other cases that hit this code are not broken.

File: .NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDateParser.cs [hunk 2]
```diff
@@ -627,7 +627,7 @@ private static DateObject ComputeDate(int cardinal, int weekday, int month, int
...
             {
                 firstWeekday = firstDay.Next((DayOfWeek)weekday);
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDateParser.cs]: Please also include the fix in javascript.

File: Specs/DateTime/English/DateParser.json [hunk 3]
```diff
@@ -2355,4 +2355,76 @@
     }
   ]
 }
+,
+{
+  "Input": "I'll see you the first Friday of next month",
```
Comment (tellarin) [inline Specs/DateTime/English/DateParser.json]: Can you add tests for “first Thursday of next year”?
Also, there are already tests for “Tuesday next week”, “Sunday of next week”, and similar, right?

File: Specs/DateTime/English/DateParser.json [hunk 4]
```diff
@@ -2355,4 +2355,76 @@
...
+  "Context": {
+    "ReferenceDateTime": "2018-03-20T09:58:00"
+  },
+  "NotSupported": "javascript",
```
Comment (tellarin) [inline Specs/DateTime/English/DateParser.json]: If you fixed it in JavaScript, please remove the NotSupported: javascript tags from the test specs.

File: Specs/DateTime/English/DateParser.json (hunk 3 above)
Comment (AniChikage) [reply Specs/DateTime/English/DateParser.json]: “first Thursday of next year” is not recognized, I'll fix it.
“Tuesday next week”, “Sunday of next week” are supported, I'll add some tests for them.

File: Specs/DateTime/English/DateParser.json (hunk 4 above)
Comment (AniChikage) [reply Specs/DateTime/English/DateParser.json]: OK, I'll remove it.

File: .NET/CreateAllPackages.cmd [hunk 5]
```diff
@@ -30,7 +30,7 @@ popd
 pushd Microsoft.Recognizers.Text.Choice
 call CreatePackage.cmd
 popd
-pushd Microsoft.Recognizers.Text.DataTypes.TimexExpression
```
Comment (tellarin) [inline .NET/CreateAllPackages.cmd]: Please don’t revert other people changes. You should rebase/merge your PR on /master

**Comment by AniChikage** [comment]
This commit only fixes bug "first Friday of next month" returning Friday of next week, JS is corrected accordingly. Some basic tests are included in Specs, and have passed.

**Pull request merged**